### ISSUE TITLE
Fully detach single field from ODocument

### ISSUE THREAD

**Issue by jdillon** [orientechnologies/orientdb#3433]
I would like to be able to fully detach (and divorce from all orientdb structures) a single field from an ODocument.

```
Map<String, Map<String, Object>> attributes = document.field("attributes", OType.EMBEDDEDMAP);
```

If I `document.detach()` I still see that the map contains ODocument values.

Would really like to have some API here, maybe on ODocument or ODocumentHelper to fully detach a single field.

```
Map<String, Map<String, Object>> attributes = ODocumentHelper.fullyDetachField(document, "attributes", OType.EMBEDDEDMAP);
```

... or something like that, so that the `attributes` variable contains just basic values an collections, etc.

ATM I have something which does this manually, but its likely missing some types and cases and its not very pretty:

```
  //
  // Object copy helpers
  //

  private Object maybeCopy(final Object source) {
    if (source instanceof ODocument) {
      return copy((ODocument)source);
    }
    else if (source instanceof Map) {
      return copy((Map) source);
    }
    else if (source instanceof List) {
      return copy((List) source);
    }
    else if (source instanceof Set) {
      return copy((Set)source);
    }
    return source;
  }

  @SuppressWarnings("unchecked")
  protected Map copy(final ODocument source) {
    String[] names = source.fieldNames();
    Map target = Maps.newHashMapWithExpectedSize(names.length);
    for (String name : names) {
      Object value = maybeCopy(source.field(name));
      target.put(name, value);
    }
    return target;
  }

  @SuppressWarnings("unchecked")
  protected Map copy(final Map<?, ?> source) {
    Map target = Maps.newHashMapWithExpectedSize(source.size());
    for (Map.Entry<?, ?> entry : source.entrySet()) {
      Object value = maybeCopy(entry.getValue());
      target.put(entry.getKey(), value);
    }
    return target;
  }

  @SuppressWarnings("unchecked")
  protected List copy(final List<?> source) {
    List target = Lists.newArrayListWithExpectedSize(source.size());
    for (Object value : source) {
      value = maybeCopy(value);
      target.add(value);
    }
    return target;
  }

  @SuppressWarnings("unchecked")
  protected Set copy(final Set<?> source) {
    Set target = Sets.newHashSetWithExpectedSize(source.size());
    for (Object value : source) {
      value = maybeCopy(value);
      target.add(value);
    }
    return target;
  }
```

And then:

```
Map<String, Map<String, Object>> attributes = document.field("attributes", OType.EMBEDDEDMAP);
if (attributes != null) {
    attributes = copy(attributes);
}
```

It would be nice if there was a supported helper api in orient which could do this for me.
